File: frontend/src/types/portfolio.ts
/**
 * 포트폴리오 관련 타입 정의
 */

export interface Position {
  symbol: string
  amount: number
  entry_price: number
  current_price: number
  unrealized_pnl: number
  stop_loss?: number
  take_profit?: number
  /** 봇이 진입·추적 중인 포지션 */
  managed_by_bot?: boolean
}

export interface Portfolio {
  total_value_usd: number
  cash_usd: number
  positions: Record<string, Position>
  pnl_today: number
  pnl_total: number
  total_trades: number
  winning_trades: number
  losing_trades: number
  win_rate: number
  total_return_pct: number
  initial_balance: number
  updated_at: string
  /** 실거래: 거래소 현물 잔고 동기화 */
  live_trading?: boolean
  data_source?: 'exchange' | 'simulated'
}

File: backend/schemas/portfolio.py
"""포트폴리오 관련 Pydantic 스키마"""
from datetime import datetime
from typing import Dict, Optional
from pydantic import BaseModel


class PositionResponse(BaseModel):
    """보유 포지션 응답"""
    symbol: str
    amount: float
    entry_price: float
    current_price: float
    unrealized_pnl: float
    stop_loss: Optional[float] = None
    take_profit: Optional[float] = None


class PortfolioResponse(BaseModel):
    """포트폴리오 전체 응답"""
    total_value_usd: float
    cash_usd: float
    positions: Dict[str, dict]
    pnl_today: float
    pnl_total: float
    total_trades: int
    winning_trades: int
    losing_trades: int
    win_rate: float
    total_return_pct: float
    initial_balance: float
    updated_at: datetime
    live_trading: bool = False
    data_source: str = "simulated"  # exchange | simulated
